pkg/cmd/gen_driftignore.go

```diff
@@ -16,19 +17,34 @@ func NewGenDriftIgnoreCmd() *cobra.Command {
 	cmd := &cobra.Command{
 		Use:   "gen-driftignore",
 		Short: "Generate a .driftignore file based on your scan result",
-		Long:  "This command will generate a new .driftignore file containing your current drifts and send output to /dev/stdout\n\nExample: driftctl scan -o json://stdout | driftctl gen-driftignore -i /dev/stdin > .driftignore",
+		Long:  "This command will generate a new .driftignore file containing your current drifts\n\nExample: driftctl scan -o json://stdout | driftctl gen-driftignore",
 		Args:  cobra.NoArgs,
 		RunE: func(cmd *cobra.Command, args []string) error {
-			if opts.InputPath == "" {
-				return errors.New("Error: you must specify an input to parse JSON from. Use driftctl gen-driftignore -i <drifts.json>\nGenerate a JSON file using the output flag: driftctl scan -o json://path/to/drifts.json")
+			driftFile := os.Stdin
+			if opts.InputPath != "-" {
+				var err error
+				driftFile, err = os.Open(opts.InputPath)
+				if err != nil {
+					return err
+				}
+				defer driftFile.Close()
```

Comment:
Why not putting this logic directly in `genDriftIgnore` ? I do not see any reason to keep it here.

For me the `RunE` from a command should be viewed as Controller in an MVP approach, and we want to keep less business as possible here. It will be also easier to unit test `genDriftIgnore`, I know that we are testing the whole cmd run but I like to keep things separated. This logic for me is a part of the business logic and the responsibility of `genDriftIgnore`

I'm wondering if we should move the output logic within too 🤔 For me this is fine to have output logic and formatting stuff in a controller but I hesitate. WDYT @craigfurman ?

Reply:
I agree with the analogy to controllers in web-type-things, and that in general you don't want to mix a ton of logic in there, for sanity's sake 😅 

This is quite subjective, but the yardstick I normally use for what to put "at the edge", in handlers for CLI / web adapters, and what to put deeper in the core, is: map inputs and outputs to whatever technical domain\* we're targeting in the "controller", and put the rest in the core, as long as that core function can be written with a "pure go" signature that has no / very-little dependency on the technical details of how the program does I/O.

But, having said all that - now that I re-look at the code, `genDriftIgnore` is just a function in the same file! I'd be totally happy to move this part.

\*: by "technical domain", I mean things like parsing flags and writing to files for a CLI like driftctl, or parsing HTTP requests and serialising responses for a web handler.

Reply:
> But, having said all that - now that I re-look at the code, genDriftIgnore is just a function in the same file! I'd be totally happy to move this part.

Oh you are right so no big deal here 😅 I've made expectations on the partial diff output I was reading without looking deeper 🙏🏻